rfcs/2021-06-15-shipping-blade.md

```diff
@@ -0,0 +1,355 @@
+
+## How to package?
+We have different tools at hand to package.
+* Babel: It'll just transpile the code and won't bundle things
+* Rollup: It'll bundle the code into one single file/multiple(depending on our config)
+* Parcel: It'll bundle the code into one single file/multiple(depending on our config)
+* Bob: This under the hood transpiles things using babel and is more suitable for complex libraries built for react native apps and esp ones interacting with native side(android/iOS). Don't yet see a use case for this so not considering it at the moment.
+
+### Shall we Transpile(using babel)?
+It'll just transpile the code and won't bundle things. Before we take a call let's see a checklist of what transpiling can/can't do.
+
+1. We can't use library with the script tag in html(umd bundle).
+    ```html
+    <script src="https://unpkg.com/@razorpay/blade@0.1.1/index.js" crossorigin></script>
+    ```
+2. Consumers are allwoed to import anything from anywhere.
+    ```js
+    // this is allowed
+    import { Button } from '@razorpay/blade'
+    // this is allowed too
+    import Button from '@razorpay/blade/_SomeInternalButton'
+    ```
+3. As library authors we are not responsible for tree shaking, we depend on the consumer applications to take care of that. We'll just give them transpiled code so their app tools can understand.
+4. Working of library totally depends on the tools config and different versions of consumer apps. It should at least have the configs(babel presets/plugins) in which the library is built.
+> 📝 We can create our custom plugins and ask consumers to import it but that's again too much effort to maintain another piece.
+
+### Shall we Bundle(using rollup/similar tools)?
+It'll bundle the code into one single file/multiple(depending on our config).Before we take a call let's see a checklist of what transpiling can/can't do.
+1. We can use library with the script tag in html(umd bundle).
+    ```html
+    <script src="https://unpkg.com/@razorpay/blade@0.1.1/index.js" crossorigin></script>
+    ```
+2. The imports are restricted to what we re-export and bundle.
+    ```js
+    // this is allowed
+    import { Button } from '@razorpay/blade'
+    // this is not allowed as it's not re-exported
+    import Button from '@razorpay/blade/_SomeInternalButton'
+    ```
+3. As library authors we are responsible for tree shaking since we'll be bundling the code which will be ready to be used by the consumers.
+4. Since we ship the packaged code we don't care what tools config or versions the consumer app is using. We'll ship code that they just need to plug in and it would work.
+
+### Conclusion
+I feel we below is the checklist that we need.
+
+✅ Can use library with the script tag in html(umd bundle).
+
+✅ Library author responsible for tree shaking
+
+❌ Import anything from anywhere
```

Comment:
Confused about what you meant here?

Is it:-

**We dont need this item: `Import anything from anywhere`**

or does it mean

**We should not allow import anything from anywhere (as you mentioned above in some section)**

If its the second I guess change the emoji and rephrase the statements.